tests/common/test_cgroupapi.py

```diff
@@ -303,7 +303,8 @@ def test_it_should_return_extension_slice_name(self):
         self.assertEqual(extension_slice_name, "system-walinuxagent.extensions-Microsoft.Azure.DummyExtension_1.0.slice")
 
     @skip_if_predicate_false(i_am_root, "Test does not run when normal user")
-    def test_if_extensions_root_slice_is_created(self):
+    @attr(permissions='root')
```

Comment:
Let's remove the skip on i_am_root, or convert it to an assertion within the test. This way people running the tests on their dev machines are notified that the test needs to run as root (instead of simply skipping it, which may go unnoticed)

Reply:
When running the test in PyCharm or from command line, if the test was skipped, it will be reported as skipped.

If we remove the decorator (and add the assertion within the test), those tests will fail when invoked as a normal user. So, if we don't want to see failures in Travis, we'd have to modify the nosetest call to exclude the tests with the root attribute. The same applies when running the tests locally, which seems like an unnecessary step each time we run tests.

I'm in favour of keeping the decorator because it's very clear which tests are skipped instead of having those tests just fail and potentially add confusion as to why they are failing.

Reply:
Discussed offline, in favour of making failures explicit instead of skipped, adding the assertion if root within the tests and changing the invocation for non-root and root tests in the .yml file.